IdentityCore/src/cache/accessor/MSIDAccountCredentialCache.m

```diff
@@ -313,6 +313,51 @@ - (BOOL)removeCredentialsWithQuery:(nonnull MSIDDefaultCredentialCacheQuery *)ca
                                 error:error];
 }
 
+// Remove credentials
+- (BOOL)removeExpiredAccessTokensCredentialsWithQuery:(nonnull MSIDDefaultCredentialCacheQuery *)cacheQuery
+                          context:(nullable id<MSIDRequestContext>)context
+                            error:(NSError * _Nullable * _Nullable)error
+{
+    assert(cacheQuery);
+    cacheQuery.targetMatchingOptions = MSIDAny;
+    cacheQuery.matchAnyCredentialType = NO;
+    NSDate *currentDataAndTime = [NSDate date];
+    NSMutableArray *resultsToDelete = [NSMutableArray array];
+    NSString *className = NSStringFromClass(self.class);
+
+    MSID_LOG_WITH_CTX_PII(MSIDLogLevelInfo, context, @"(Default cache) Removing expired credentials with type %@, environment %@, realm %@, clientID %@, unique user ID %@", [MSIDCredentialTypeHelpers credentialTypeAsString:cacheQuery.credentialType], cacheQuery.environment, cacheQuery.realm, cacheQuery.clientId, MSID_PII_LOG_TRACKABLE(cacheQuery.homeAccountId));
+
+    if (cacheQuery.exactMatch)
+    {
+        return NO;
+    }
+    
+    if (cacheQuery.credentialType != MSIDAccessTokenType)
+    {
+        return NO;
+    }
+
+    NSArray<MSIDCredentialCacheItem *> *matchedCredentials = [self getCredentialsWithQuery:cacheQuery context:context error:error];
+    
+    if (!matchedCredentials) return NO;
```

Comment:
Change to if (!matchedCredentials.count) return NO; to check for empty array

Reply:
The same pattern is used removeCredentialsWithQuery above. It is a null check. getCredentialsWithQuery returns nil on errors.